src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,111 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+type Movies = {
+  onAdd: (film: Movie) => void;
+};
+
+export const NewMovie: React.FC<Movies> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [newMovie, setNewMovie] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+
+  const {
+    title, description, imgUrl, imdbUrl, imdbId,
+  } = newMovie;
+
+  const [count, setCount] = useState(0);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+
+    setNewMovie((prevMovie) => ({ ...prevMovie, [name]: value }));
+  };
+
+  const buttonAction = !title || !imgUrl || !imdbUrl || !imdbId;
```

Comment:
```suggestion
  const buttonAction = !title.trim() || !imgUrl.trim() || !imdbUrl.trim() || !imdbId.trim();
```

it will fix the bug with an empty movie